### PR TITLE
Enable travis, etc

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+language: dart
+sudo: false
+dart:
+  - dev
+  - stable
+  - 1.24.2
+  - 1.23.0
+  - 1.22.1
```
Comment (nex3) [inline .travis.yml]: I don't think it's worth testing more than dev or stable here. It'll just add to the iteration time without any substantial gain.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+dart_task:
+  - test: --platform vm
+  - test: --platform firefox
+  - test: --platform dartium
+    install_dartium: true
+
+matrix:
+  allow_failures:
+    - dart: dev
+      dart_task: dartfmt
```
Comment (nex3) [inline .travis.yml]: Why is this in `allow_failures`?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+language: dart
+sudo: false
+dart:
+  - dev
+  - stable
+  - 1.24.2
+  - 1.23.0
+  - 1.22.1
+
+dart_task:
```
Comment (nex3) [inline .travis.yml]: It's probably overkill to run these in three separate builders; this package doesn't have that many tests.

File: analysis_options.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+analyzer:
+  strong-mode: true
+  errors:
+    unused_import: error
+    unused_local_variable: error
+    dead_code: error
+linter:
+  rules:
```
Comment (nex3) [inline analysis_options.yaml]: Lints give me hives; they aren't well-curated or -maintained in general. If I were very confident in an individual lint I'd be okay with enabling it, but this scattershot approach makes me worried that we'll end up chasing down false positives and bugs.

File: analysis_options.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+analyzer:
+  strong-mode: true
+  errors:
+    unused_import: error
+    unused_local_variable: error
+    dead_code: error
```
Comment (nex3) [inline analysis_options.yaml]: Why are these upgraded to errors?

**Review by nex3 — CHANGES_REQUESTED** [review]

File: analysis_options.yaml (hunk 5 above)
Comment (kevmoo) [reply analysis_options.yaml]: They cause errors in Google3 – staying consistent.

File: analysis_options.yaml (hunk 4 above)
Comment (kevmoo) [reply analysis_options.yaml]: These are the ones we use on angular – and google3.

They are used a lot internally. Pretty standard set.

File: .travis.yml (hunk 1 above)
Comment (kevmoo) [reply .travis.yml]: Fixed

File: .travis.yml (hunk 3 above)
Comment (kevmoo) [reply .travis.yml]: With a smaller matrix, it makes debugging pretty easy.

**Comment by kevmoo** [comment]
@nex3 PTAL?

File: test/current_isolate_info_test.dart [hunk 6]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
+@TestOn('vm')
```
Comment (nex3) [inline test/current_isolate_info_test.dart]: Nit: newline above

File: analysis_options.yaml (hunk 5 above)
Comment (nex3) [reply analysis_options.yaml]: Why does Google3 config control which hints are upgraded in third-party packages?

File: analysis_options.yaml (hunk 4 above)
Comment (nex3) [reply analysis_options.yaml]: That's fine for them, but I'd prefer to leave these off of my packages.

File: test/package_config_info_test.dart [hunk 7]
```diff
@@ -157,7 +160,7 @@ void main() {
 
     test("with a matching package", () {
       expect(resolver.packagePath("foo"), equals(p.current));
-    });
+    }, onPlatform: {'!vm': new Skip()});
```
Comment (nex3) [inline test/package_config_info_test.dart]: You should never commit a skip without a description.

If this test isn't compatible with the browser, you should use `testOn: "vm"`, not a skip. Skips are for tests that are broken and need to be fixed.

Same below.

**Review by nex3 — CHANGES_REQUESTED** [review]

**Comment by kevmoo** [comment]
`runOn` fixed. 

Lints removed – although they are a great way to enforce style and to avoid common mistakes in code. They also help us guide contributors through PRs by highlighting common mistakes.

File: .travis.yml (hunk 3 above)
Comment (nex3) [reply .travis.yml]: That's true, but it also decreases throughput and soaks up builders that other packages may need.

**Pull request merged**